utility/src/test/java/org/oppia/android/util/networking/DebugNetworkConnectionUtilTest.kt

```diff
@@ -0,0 +1,278 @@
+      status = ConnectivityManager.TYPE_MOBILE,
+      networkState = NetworkInfo.State.DISCONNECTED
+    )
+    debugNetworkConnectionUtil.setCurrentConnectionStatus(ConnectionStatus.NONE)
+    assertThat(debugNetworkConnectionUtil.getCurrentConnectionStatus()).isEqualTo(
+      ConnectionStatus.NONE
+    )
+  }
+
+  @Test
+  fun testGetCurrentConnectionStatus_nonActiveCellular_noForcedConnection_returnsNone() {
+    setNetworkConnectionStatus(
+      status = ConnectivityManager.TYPE_MOBILE,
+      networkState = NetworkInfo.State.DISCONNECTED
+    )
+    assertThat(debugNetworkConnectionUtil.getCurrentConnectionStatus()).isEqualTo(
+      ConnectionStatus.NONE
+    )
+  }
+
+  @Test
+  fun testGetCurrentConnectionStatus_nonActiveCellular_forceWifiConnection_returnsNone() {
+    setNetworkConnectionStatus(
+      status = ConnectivityManager.TYPE_MOBILE,
+      networkState = NetworkInfo.State.DISCONNECTED
+    )
+    debugNetworkConnectionUtil.setCurrentConnectionStatus(ConnectionStatus.LOCAL)
+    assertThat(debugNetworkConnectionUtil.getCurrentConnectionStatus()).isEqualTo(
+      ConnectionStatus.NONE
+    )
+  }
+
+  @Test
+  fun testGetCurrentConnectionStatus_nonActiveCellular_forceCellularConnection_returnsNone() {
+    setNetworkConnectionStatus(
+      status = ConnectivityManager.TYPE_MOBILE,
+      networkState = NetworkInfo.State.DISCONNECTED
+    )
+    debugNetworkConnectionUtil.setCurrentConnectionStatus(ConnectionStatus.CELLULAR)
+    assertThat(debugNetworkConnectionUtil.getCurrentConnectionStatus()).isEqualTo(
+      ConnectionStatus.NONE
+    )
+  }
+
+  @Test
+  fun testGetCurrentConnectionStatus_noActiveNetwork_noForcedConnection_returnsNone() {
+    setNetworkConnectionStatus(
+      status = NO_CONNECTION,
+      networkState = NetworkInfo.State.DISCONNECTED
+    )
+    assertThat(debugNetworkConnectionUtil.getCurrentConnectionStatus()).isEqualTo(
+      ConnectionStatus.NONE
+    )
+  }
+
+  @Test
+  fun testGetCurrentConnectionStatus_noActiveNetwork_forceWifiConnection_returnsNone() {
+    setNetworkConnectionStatus(
+      status = NO_CONNECTION,
+      networkState = NetworkInfo.State.DISCONNECTED
+    )
+    debugNetworkConnectionUtil.setCurrentConnectionStatus(ConnectionStatus.LOCAL)
+    assertThat(debugNetworkConnectionUtil.getCurrentConnectionStatus()).isEqualTo(
+      ConnectionStatus.NONE
+    )
+  }
+
+  @Test
+  fun testGetCurrentConnectionStatus_noActiveNetwork_forceCellularConnection_returnsNone() {
+    setNetworkConnectionStatus(
+      status = NO_CONNECTION,
+      networkState = NetworkInfo.State.DISCONNECTED
+    )
+    debugNetworkConnectionUtil.setCurrentConnectionStatus(ConnectionStatus.CELLULAR)
+    assertThat(debugNetworkConnectionUtil.getCurrentConnectionStatus()).isEqualTo(
+      ConnectionStatus.NONE
+    )
+  }
+
+  private fun setNetworkConnectionStatus(status: Int, networkState: NetworkInfo.State) {
+    NetworkConnectionTestUtil.setNetworkInfo(context, status, networkState)
+  }
```

Comment:
Per later comments on making this utility injectable, I think this helper function can eventually be removed here & in other tests where applicable.

Reply:
Done.